django/core/cache/backends/base.py

```diff
@@ -160,16 +160,15 @@ def get_or_set(self, key, default, timeout=DEFAULT_TIMEOUT, version=None):
 
         Return the value of the key stored or retrieved.
         """
-        val = self.get(key, version=version)
-        if val is None:
+        val = self.get(key, default, version=version)
+        if val in [None, default]:
```

Comment:
I'm not understanding why we enter this branch if `val` is `default`. Would you be able to explain? Thanks!